### PR TITLE
5093 Jump functionality for ec and g2c

### PR DESCRIPTION
**Pull request by BarinderBanwait**
This addresses #5093 

Behavior is live at [pink](https://pink.lmfdb.xyz/)

Some comments about code:

1. Relevant regular expressions have been moved to a new file in the `utils` folder (as agreed during a Friday meeting) since both ecQ and g2c need them.
2. Modest testing added for both jumps.

Have a play with the pink branch and let me know of any errors or changes I should make. Meanwhile I'll fix the obligatory test failures as they arise.

### CODE REVIEW

**Comment by BarinderBanwait** [comment]
@roed314 I'm puzzled why 

https://www.lmfdb.org/EllipticCurve/Q/?jump=%5B-3024%2C+46224%5D+

works; but typing `[-3024,+46224]+` into the elliptic curve jump does not? Shouldn't doing the latter also give the same jump link? This is related to an existing test (`test_jump` in `test_browse_page.py` in elliptic_curves).

**Comment by BarinderBanwait** [comment]
Succès!!!!! @roed314 @edgarcosta

File: lmfdb/elliptic_curves/elliptic_curve.py [hunk 1]
```diff
@@ -257,49 +260,114 @@ def elliptic_curve_jump_error(label, args, missing_curve=False, missing_class=Fa
...
+    def read_list_coeffs(elt):
+        if not elt:
+            return R(0)
+        else:
+            return R([int(c) for c in elt.split(",")])
+
+    if ZLLIST_RE.fullmatch(input_str):
+        input_str_new = input_str.strip('[').strip(']')
+        fg = [read_list_coeffs(elt) for elt in input_str_new.split('],[')]
+    # elif ZLIST_RE.fullmatch(input_str):
```
Comment (roed314) [inline lmfdb/elliptic_curves/elliptic_curve.py:305]: These should be removed

File: lmfdb/elliptic_curves/elliptic_curve.py [hunk 2]
```diff
@@ -257,49 +260,114 @@ def elliptic_curve_jump_error(label, args, missing_curve=False, missing_class=Fa
...
+    #     input_str_new = input_str.strip('[').strip(']')
+    #     fg = [read_list_coeffs(input_str_new), R(0)]
+    else:
+        input_str_new = input_str.strip('[').strip(']')
+        fg = [R(list(coeff_to_poly(elt))) for elt in input_str_new.split(",")]
+    if len(fg) == 1:
+        fg.append(R(0))
+
+    ec_defining_poly = y**2 + y*(fg[1]) - fg[0]
+    S = PolynomialRing(QQ,2, ["x", "y"])
```
Comment (roed314) [inline lmfdb/elliptic_curves/elliptic_curve.py:315]: spaces after commas.

File: lmfdb/elliptic_curves/elliptic_curve.py (hunk 1 above)
Comment (BarinderBanwait) [reply lmfdb/elliptic_curves/elliptic_curve.py]: done in latest commit

File: lmfdb/elliptic_curves/elliptic_curve.py (hunk 2 above)
Comment (BarinderBanwait) [reply lmfdb/elliptic_curves/elliptic_curve.py]: done in latest commit

**Pull request merged**